### DISCUSSION TITLE
Puppet learning VM compatible with Apple Silicon

### DISCUSSION THREAD

**Discussion by Udemon** [puppetlabs/community#19] [Xtra Everything Else]
Has anyone managed to set up a puppet learning VM on a Mac with M1/M1 pro?

**Comment by BryanGuffeySonos** [comment]
Does anyone have any info on this

**Comment by jay7x** [comment]
I've just tried to run the `learning_puppet_vm-6.11` under the `qemu-system-x86_64` on my Macbook with M1 Pro..
Though it doesn't really works.. QEMU is still slow so the `sys` time inside of the VM is very high.. that caused very high LA.. so systemd services are timing out on start.

Below is what I did:

```bash
wget https://pe-education-vms.s3.amazonaws.com/learning/learning_puppet_vm-6.11.zip
unzip learning_puppet_vm-6.11.zip
tar -xf puppet-2019.8.1-learning-6.11.ova
qemu-img convert -p -c -O qcow2 learning-disk001.vmdk learning-base.qcow2
rm -f learning-disk001.qcow2
qemu-img create -f qcow2 -b learning-base.qcow2 -F qcow2 learning-disk001.qcow2

qemu-system-x86_64 \
-accel tcg,thread=multi \
-cpu qemu64 \
-smp 8 \
-m 6G \
-netdev user,\
hostfwd=tcp::2222-:22,\
hostfwd=tcp::8080-:80,\
hostfwd=tcp::8443-:443,\
id=net0 \
-device virtio-net,netdev=net0 \
-drive if=virtio,format=qcow2,file=learning-disk001.qcow2 \
-display cocoa
```

I'll try to increase systemd timeout but I don't think it'll really help.. better to rebuild the VM on any other arm64 RHEL-based distro which has puppet-agent released for..

P.S. Note that `-accel tcg,thread=multi` can cause some memory-related issues.. not sure if JVM is ok with this. Though QEMU is insanely slow with single-threaded `tcg`..
